Updated documentation for OshiSysMonitor (#14799)
apache/druid: docs/operations/metrics.md

  -->

> You can configure Druid to [emit metrics](../configuration/index.md#enabling-metrics) that are essential for monitoring

Reviewer: ```suggestion
You can configure Druid to [emit metrics](../configuration/index.md#enabling-metrics) that are essential for monitoring query execution, ingestion, coordination, and so on.
```

> query execution, ingestion, coordination, and so on.

Reviewer: Remove line 27.

## Query metrics

### Router

> |Metric|Description|Dimensions|Normal Value|

Reviewer: ```suggestion
|Metric|Description|Dimensions|Normal value|
```

#### Memcached only metrics

> Memcached client metrics are reported as per the following. These metrics come directly from the client as opposed to

Reviewer: ```suggestion
Memcached client metrics are reported as per the following. These metrics come directly from the client.
```

> from the cache retrieval layer.

Reviewer: Remove line 141.

|`ingest/segments/count`|Count of final segments created by job (includes tombstones). | `dataSource`, `taskId`, `taskType`, `groupId`, `taskIngestionMode`, `tags` |At least `1`.|
|`ingest/tombstones/count`|Count of tombstones created by job. | `dataSource`, `taskId`, `taskType`, `groupId`, `taskIngestionMode`, `tags` |Zero or more for replace. Always zero for non-replace tasks (always zero for legacy replace, see below).|

The `taskIngestionMode` dimension includes the following modes:

> 

Author: This blank line is coming again with auto formatting. Do you want me to remove this ?

* `APPEND`: a native ingestion job appending to existing segments
* `REPLACE_LEGACY`: the original replace before tombstones
* `REPLACE`: a native ingestion job replacing existing segments using tombstones

The mode is decided using the values
of the `isAppendToExisting` and `isDropExisting` flags in the
task's `IOConfig` as follows:

> | `isAppendToExisting` | `isDropExisting` | mode |

Reviewer: ```suggestion
|`isAppendToExisting`|`isDropExisting`|mode|
```

> |----------------------|-------------------|------|

Reviewer: ```suggestion
|--------------------|----------------|----|
```

> | `true`               | `false` | `APPEND`|

Reviewer: ```suggestion
|`true`|`false`|`APPEND`|
```

> | `true`               | `true  ` | Invalid combination, exception thrown. |

Reviewer: ```suggestion
|`true`|`true`|Invalid combination, exception thrown.|
```

> | `false`              | `false` | `REPLACE_LEGACY` (this is the default for native batch ingestion). |

Reviewer: ```suggestion
|`false`|`false`|`REPLACE_LEGACY` (This is the default for native batch ingestion.)|
```

> | `false`              | `true` | `REPLACE`|

Reviewer: ```suggestion
|`false`|`true`|`REPLACE`|
```

|`ingest/notices/queueSize`|Number of pending notices to be processed by the coordinator.|`dataSource`, `tags`|Typically 0 and occasionally in lower single digits. Should not be a very high number. |
|`ingest/notices/time`|Milliseconds taken to process a notice by the supervisor.|`dataSource`, `tags`| < 1s |
|`ingest/pause/time`|Milliseconds spent by a task in a paused state without ingesting.|`dataSource`, `taskId`, `tags`| < 10 seconds|

> |`ingest/handoff/time`|Total number of milliseconds taken to handoff a set of segments.|`dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Depends on coordinator cycle time.|

Reviewer: ```suggestion
|`ingest/handoff/time`|Total number of milliseconds taken to handoff a set of segments.|`dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Depends on the coordinator cycle time.|
```

## Shuffle metrics (Native parallel task)

> The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor`

Reviewer: ```suggestion
The shuffle metrics can be enabled by adding `org.apache.druid.indexing.worker.shuffle.ShuffleMonitor` in `druid.monitoring.monitors`.
```

> in `druid.monitoring.monitors`

Reviewer: Remove line 288.

> in `druid.monitoring.monitors`

Reviewer: ```suggestion
in `druid.monitoring.monitors`.
```

> See [Enabling Metrics](../configuration/index.md#enabling-metrics) for more details.

Reviewer: ```suggestion
See [Enabling metrics](../configuration/index.md#enabling-metrics) for more details.
```

|`ingest/events/buffered`|Number of events queued in the `EventReceiverFirehose` buffer.|`serviceName`, `dataSource`, `taskId`, `taskType`, `bufferCapacity`|Equal to the current number of events in the buffer queue.|
|`ingest/bytes/received`|Number of bytes received by the `EventReceiverFirehose`.|`serviceName`, `dataSource`, `taskId`, `taskType`|Varies|

## Sys [Deprecated]

> > SysMonitor will be deprecated due to its reliance on the sigar library, which is no longer maintained.

Reviewer: ```suggestion
> SysMonitor will be deprecated due to its reliance on the Sigar library, which is no longer maintained.
```

> > A new Monitor called [OshiSysMonitor](#oshisysmonitor), which provides similar functionality but is based on the oshi

Reviewer: ```suggestion
> Instead, use the new OSHI monitor called [OshiSysMonitor](#oshisysmonitor). The new monitor has a wider support for different machine architectures including ARM instances.
```

> > library. We encourage all users to switch to OshiSysMonitor as soon as possible.

Reviewer: Remove line 415.

These metrics are only available if the `SysMonitor` module is included.

> | Metric                 | Description                                                                          |Dimensions|Normal Value|

Reviewer: If `SysMonitor` is deprecated, should these metrics be removed?

Also, please fix the table formatting. Do not add whitespace to make columns longer. Trim excessive white space between columns.

| `sys/net/read/size`    | Bytes read from the network                                                          |`netName`, `netAddress`, `netHwaddr`|Varies|
| `sys/fs/used`          | Filesystem bytes used                                                                |`fsDevName`, `fsDirName`, `fsTypeName`, `fsSysTypeName`, `fsOptions`|< max|
| `sys/fs/max`           | Filesystem bytes max                                                                 |`fsDevName`, `fsDirName`, `fsTypeName`, `fsSysTypeName`, `fsOptions`|Varies|
| `sys/mem/used`         | Memory used                                                                          ||< max|
| `sys/mem/max`          | Memory max                                                                           ||Varies|
| `sys/storage/used`     | Disk space used                                                                      |`fsDirName`|Varies|
| `sys/cpu`              | CPU used                                                                             |`cpuName`, `cpuTime`|Varies|

## OshiSysMonitor

These metrics are only available if the `OshiSysMonitor` module is included.

> | Metric                     | Description                                                                          | Dimensions                           | Normal Value        |

Reviewer: Please fix the table formatting.

Do not add whitespace to make columns longer. Trim excessive white space between columns.
